core/src/main/scala/chisel3/Module.scala

```diff
@@ -107,6 +110,32 @@ object Module extends SourceInfoDoc {
   /** Returns the implicit Reset, if it is defined */
   def resetOption: Option[Reset] = Builder.currentReset
 
+  /** Returns the current implicit [[Disable]], if one is defined
+    *
+    * Note that [[Disable]] is a function of the implicit clock and reset
+    * so having no implicit clock or reset may imply no `Disable`.
+    */
+  def disable(implicit sourceInfo: SourceInfo): Option[Disable] = {
+    Builder.currentDisable match {
+      case Disable.None            => None
+      case Disable.NotHasBeenReset => hasBeenReset.map(x => autoNameRecursively("disable")(!x))
+    }
+  }
+
+  // Should this be public or should users just go through .disable?
+  // Note that having a reset but not clock means hasBeenReset is None, should we default to just !reset?
+  private def hasBeenReset(implicit sourceInfo: SourceInfo): Option[Disable] = {
+    // TODO memoize this
+    (Builder.currentClock, Builder.currentReset) match {
+      case (Some(clock), Some(reset)) =>
+        val hasBeenReset = Module(new HasBeenResetIntrinsic)
+        hasBeenReset.clock := clock
+        hasBeenReset.reset := reset
+        Some(new Disable(hasBeenReset.out))
+      case _ => None
+    }
```

Comment:
Either update this with the memoization or land it before this gets in to production.

Reply:
I'll do it in a quick follow on PR, working on it.

Reply:
The memoization of the instance is one thing we need, we also need to use D/I on the Intrinsic definitions. There is a bug in ScalaDoc that is making this difficult to do but will handle it in https://github.com/chipsalliance/chisel/pull/3499.